Problem with ArgoCD Multi-Source Application Support

### Current Behavior

I am using your latest instructions here to the letter: https://github.com/jkroepke/helm-secrets/wiki/ArgoCD-Integration#multi-source-application-support-beta

I am seeing encrypted values in my Kubernetes secrets and from the `repo-server` log:

`download-tools sh: can't create /usr/local/sbin/helm: nonexistent directory`

Might there be an error on this line?

`printf '#!/usr/bin/env sh\nif [ "${HELM_SECRETS_WRAPPER_ENABLED}" = "true" ]; then exec %1$s secrets "$@"; fi\nexec %1$s "$@"' "helm" "${HELM_SECRETS_HELM_PATH}" >"/usr/local/sbin/helm" && chmod +x "/custom-tools/helm"`

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

- Helm Secrets Version: 4.4.0
- ArgoCD Version : 2.6.3

### Anything else?

_No response_

Here's the complete error message:

```ComparisonError
rpc error: code = Unknown desc = Manifest generation error (cached): `helm template . --name-template actions-runner-deployment-io --namespace actions-runner --kube-version 1.23 --values .configuration/actions-runner-deployment-io/values.vld.yaml --values .configuration/actions-runner-deployment-io/secrets.vld.enc.yaml --api-versions acme.cert-manager.io/v1 --api-versions acme.cert-manager.io/v1/Challenge --api-versions acme.cert-manager.io/v1/Order --api-versions actions.summerwind.dev/v1alpha1 --api-versions actions.summerwind.dev/v1alpha1/HorizontalRunnerAutoscaler --api-versions actions.summerwind.dev/v1alpha1/Runner --api-versions actions.summerwind.dev/v1alpha1/RunnerDeployment --api-versions actions.summerwind.dev/v1alpha1/RunnerReplicaSet --api-versions actions.summerwind.dev/v1alpha1/RunnerSet --api-versions admissionregistration.k8s.io/v1 --api-versions admissionregistration.k8s.io/v1/MutatingWebhookConfiguration --api-versions admissionregistration.k8s.io/v1/ValidatingWebhookConfiguration --api-versions apiextensions.k8s.io/v1 --api-versions apiextensions.k8s.io/v1/CustomResourceDefinition --api-versions apiregistration.k8s.io/v1 --api-versions apiregistration.k8s.io/v1/APIService --api-versions apps/v1 --api-versions apps/v1/ControllerRevision --api-versions apps/v1/DaemonSet --api-versions apps/v1/Deployment --api-versions apps/v1/ReplicaSet --api-versions apps/v1/StatefulSet --api-versions argoproj.io/v1alpha1 --api-versions argoproj.io/v1alpha1/AppProject --api-versions argoproj.io/v1alpha1/Application --api-versions argoproj.io/v1alpha1/ApplicationSet --api-versions argoproj.io/v1alpha1/ClusterWorkflowTemplate --api-versions argoproj.io/v1alpha1/CronWorkflow --api-versions argoproj.io/v1alpha1/EventBus --api-versions argoproj.io/v1alpha1/EventSource --api-versions argoproj.io/v1alpha1/Sensor --api-versions argoproj.io/v1alpha1/Workflow --api-versions argoproj.io/v1alpha1/WorkflowArtifactGCTask --api-versions argoproj.io/v1alpha1/WorkflowEventBinding --api-versions argoproj.io/v1alpha1/WorkflowTaskResult --api-versions argoproj.io/v1alpha1/WorkflowTaskSet --api-versions argoproj.io/v1alpha1/WorkflowTemplate --api-versions autoscaling/v1 --api-versions autoscaling/v1/HorizontalPodAutoscaler --api-versions autoscaling/v2 --api-versions autoscaling/v2/HorizontalPodAutoscaler --api-versions autoscaling/v2beta1 --api-versions autoscaling/v2beta1/HorizontalPodAutoscaler --api-versions autoscaling/v2beta2 --api-versions autoscaling/v2beta2/HorizontalPodAutoscaler --api-versions batch/v1 --api-versions batch/v1/CronJob --api-versions batch/v1/Job --api-versions batch/v1beta1 --api-versions batch/v1beta1/CronJob --api-versions cert-manager.io/v1 --api-versions cert-manager.io/v1/Certificate --api-versions cert-manager.io/v1/CertificateRequest --api-versions cert-manager.io/v1/ClusterIssuer --api-versions cert-manager.io/v1/Issuer --api-versions certificates.k8s.io/v1 --api-versions certificates.k8s.io/v1/CertificateSigningRequest --api-versions configuration.konghq.com/v1 --api-versions configuration.konghq.com/v1/KongClusterPlugin --api-versions configuration.konghq.com/v1/KongConsumer --api-versions configuration.konghq.com/v1/KongIngress --api-versions configuration.konghq.com/v1/KongPlugin --api-versions configuration.konghq.com/v1alpha1 --api-versions configuration.konghq.com/v1alpha1/IngressClassParameters --api-versions configuration.konghq.com/v1beta1 --api-versions configuration.konghq.com/v1beta1/TCPIngress --api-versions configuration.konghq.com/v1beta1/UDPIngress --api-versions coordination.k8s.io/v1 --api-versions coordination.k8s.io/v1/Lease --api-versions crd.k8s.amazonaws.com/v1alpha1 --api-versions crd.k8s.amazonaws.com/v1alpha1/ENIConfig --api-versions discovery.k8s.io/v1 --api-versions discovery.k8s.io/v1/EndpointSlice --api-versions discovery.k8s.io/v1beta1 --api-versions discovery.k8s.io/v1beta1/EndpointSlice --api-versions elbv2.k8s.aws/v1alpha1 --api-versions elbv2.k8s.aws/v1alpha1/TargetGroupBinding --api-versions elbv2.k8s.aws/v1beta1 --api-versions elbv2.k8s.aws/v1beta1/IngressClassParams --api-versions elbv2.k8s.aws/v1beta1/TargetGroupBinding --api-versions events.k8s.io/v1 --api-versions events.k8s.io/v1/Event --api-versions events.k8s.io/v1beta1 --api-versions events.k8s.io/v1beta1/Event --api-versions flowcontrol.apiserver.k8s.io/v1beta1 --api-versions flowcontrol.apiserver.k8s.io/v1beta1/FlowSchema --api-versions flowcontrol.apiserver.k8s.io/v1beta1/PriorityLevelConfiguration --api-versions flowcontrol.apiserver.k8s.io/v1beta2 --api-versions flowcontrol.apiserver.k8s.io/v1beta2/FlowSchema --api-versions flowcontrol.apiserver.k8s.io/v1beta2/PriorityLevelConfiguration --api-versions jaegertracing.io/v1 --api-versions jaegertracing.io/v1/Jaeger --api-versions k6.io/v1alpha1 --api-versions k6.io/v1alpha1/K6 --api-versions logging.opsgy.com/v1beta1 --api-versions logging.opsgy.com/v1beta1/GlobalLokiRule --api-versions logging.opsgy.com/v1beta1/LokiRule --api-versions monitoring.coreos.com/v1 --api-versions monitoring.coreos.com/v1/Alertmanager --api-versions monitoring.coreos.com/v1/PodMonitor --api-versions monitoring.coreos.com/v1/Probe --api-versions monitoring.coreos.com/v1/Prometheus --api-versions monitoring.coreos.com/v1/PrometheusRule --api-versions monitoring.coreos.com/v1/ServiceMonitor --api-versions monitoring.coreos.com/v1/ThanosRuler --api-versions monitoring.coreos.com/v1alpha1 --api-versions monitoring.coreos.com/v1alpha1/AlertmanagerConfig --api-versions networking.k8s.io/v1 --api-versions networking.k8s.io/v1/Ingress --api-versions networking.k8s.io/v1/IngressClass --api-versions networking.k8s.io/v1/NetworkPolicy --api-versions node.k8s.io/v1 --api-versions node.k8s.io/v1/RuntimeClass --api-versions node.k8s.io/v1beta1 --api-versions node.k8s.io/v1beta1/RuntimeClass --api-versions policy/v1 --api-versions policy/v1/PodDisruptionBudget --api-versions policy/v1beta1 --api-versions policy/v1beta1/PodDisruptionBudget --api-versions policy/v1beta1/PodSecurityPolicy --api-versions rbac.authorization.k8s.io/v1 --api-versions rbac.authorization.k8s.io/v1/ClusterRole --api-versions rbac.authorization.k8s.io/v1/ClusterRoleBinding --api-versions rbac.authorization.k8s.io/v1/Role --api-versions rbac.authorization.k8s.io/v1/RoleBinding --api-versions scheduling.k8s.io/v1 --api-versions scheduling.k8s.io/v1/PriorityClass --api-versions storage.k8s.io/v1 --api-versions storage.k8s.io/v1/CSIDriver --api-versions storage.k8s.io/v1/CSINode --api-versions storage.k8s.io/v1/StorageClass --api-versions storage.k8s.io/v1/VolumeAttachment --api-versions storage.k8s.io/v1beta1 --api-versions storage.k8s.io/v1beta1/CSIStorageCapacity --api-versions v1 --api-versions v1/ConfigMap --api-versions v1/Endpoints --api-versions v1/Event --api-versions v1/LimitRange --api-versions v1/Namespace --api-versions v1/Node --api-versions v1/PersistentVolume --api-versions v1/PersistentVolumeClaim --api-versions v1/Pod --api-versions v1/PodTemplate --api-versions v1/ReplicationController --api-versions v1/ResourceQuota --api-versions v1/Secret --api-versions v1/Service --api-versions v1/ServiceAccount --api-versions vpcresources.k8s.aws/v1beta1 --api-versions vpcresources.k8s.aws/v1beta1/SecurityGroupPolicy --include-crds` failed exit status 1: [helm-secrets] Values filepath '.configuration/actions-runner-deployment-io/values.vld.yaml' is an absolute path. Absolute paths are not allowed. [helm-secrets] File does not exist: .configuration/actions-runner-deployment-io/values.vld.yaml Error: plugin "secrets" exited with error
```

--- a/docs/ArgoCD Integration.md
+++ b/docs/ArgoCD Integration.md
@@ -52,7 +52,7 @@ spec:
         # ### Method 3: Use HELM_SECRETS_LOAD_GPG_KEYS
         # Example Method 3: (Assumptions: Pre-seeded gpg agent is running or kube service account has permission to decrypt using kms key, secrets.yaml is in the root folder)
         - secrets://secrets.yaml
-      
+
       # fileParameters (--set-file) are supported, too. 
       fileParameters:
         - name: config
@@ -71,7 +71,7 @@ because it is [registered](https://github.com/jkroepke/helm-secrets/blob/4e61c55
 
 ArgoCD has limited supported for helm-secrets and Multi-Source application. `sops` backend is not tested yet.
 
-References: 
+References:
 * https://github.com/argoproj/argo-cd/issues/11866
 * https://github.com/argoproj/argo-cd/pull/11966
 
@@ -81,8 +81,8 @@ This is in conflict with helm-secrets, since the string needs to beginn with `se
 
 `HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH` must be set to `true`, since ArgoCD pass value files with absolute file path.
 
-Ensure that the env `HELM_SECRETS_WRAPPER_ENABLED=true` (default `false`) and 
-`HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH=true` is set on the argocd-repo-server. 
+Ensure that the env `HELM_SECRETS_WRAPPER_ENABLED=true` (default `false`) and
+`HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH=true` is set on the argocd-repo-server.
 Please ensure you are following the lastest installation instructions (updated on 2023-03-03).
 
 If you are using `sops` backend, additionally define the environment variable `HELM_SECRETS_LOAD_GPG_KEYS` with the path of gpg key as values.
@@ -100,30 +100,30 @@ apiVersion: argoproj.io/v1alpha1
 kind: Application
 spec:
   sources:
-  - repoURL: 'https://prometheus-community.github.io/helm-charts'
-    chart: prometheus
-    targetRevision: 15.7.1
-    helm:
-      valueFiles:
-        # Omit any secrets:// prefix, since its not supported in multi source apps
-        - $values/charts/prometheus/secrets.yaml
+    - repoURL: 'https://prometheus-community.github.io/helm-charts'
+      chart: prometheus
+      targetRevision: 15.7.1
+      helm:
+        valueFiles:
+          # Omit any secrets:// prefix, since its not supported in multi source apps
+          - $values/charts/prometheus/secrets.yaml
 
-      # inline secret references are still supported
-      fileParameters:
-        - name: mysql.rootPassword
-          path: secrets+literal://vals!ref+awssecrets://myteam/mydoc#/foo/bar
-  - repoURL: 'https://git.example.gom/org/value-files.git'
-    targetRevision: dev
-    ref: values
+        # inline secret references are still supported
+        fileParameters:
+          - name: mysql.rootPassword
+            path: secrets+literal://vals!ref+awssecrets://myteam/mydoc#/foo/bar
+    - repoURL: 'https://git.example.gom/org/value-files.git'
+      targetRevision: dev
+      ref: values
 ```
 </p>
 </details>
 
 # Installation on Argo CD
 
-Before using helm secrets, we are required to install `helm-secrets` on the `argocd-repo-server`. 
+Before using helm secrets, we are required to install `helm-secrets` on the `argocd-repo-server`.
 Depends on the secret backend, `sops` or `vals` is required on the `argocd-repo-server`, too.
-There are two methods to do this. 
+There are two methods to do this.
 Either create your custom ArgoCD Docker Image or install them via an init container.
 
 ## Step 1: Customize argocd-repo-server
@@ -203,40 +203,50 @@ repoServer:
   env:
     - name: HELM_PLUGINS
       value: /custom-tools/helm-plugins/
+    - name: HELM_SECRETS_CURL_PATH
+      value: /custom-tools/curl
     - name: HELM_SECRETS_SOPS_PATH
       value: /custom-tools/sops
     - name: HELM_SECRETS_VALS_PATH
       value: /custom-tools/vals
     - name: HELM_SECRETS_KUBECTL_PATH
       value: /custom-tools/kubectl
-    - name: HELM_SECRETS_CURL_PATH
-      value: /custom-tools/curl
-    - name: HELM_SECRETS_HELM_PATH
-      value: /usr/local/bin/helm
     - name: HELM_SECRETS_BACKEND
-      value: "vals" # or sops
+      value: sops
     # https://github.com/jkroepke/helm-secrets/wiki/Security-in-shared-environments
     - name: HELM_SECRETS_VALUES_ALLOW_SYMLINKS
       value: "false"
     - name: HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH
-      value: "false"
+      value: "true"
     - name: HELM_SECRETS_VALUES_ALLOW_PATH_TRAVERSAL
       value: "false"
     - name: HELM_SECRETS_WRAPPER_ENABLED
-      value: "false"
+      value: "true"
+    - name: HELM_SECRETS_HELM_PATH
+      value: /usr/local/bin/helm
+
+    - name: HELM_SECRETS_LOAD_GPG_KEYS
+      # Multiple keys can be separated by space
+      value: /helm-secrets-private-keys/key.asc
   volumes:
     - name: custom-tools
       emptyDir: {}
+    # kubectl create secret generic helm-secrets-private-keys --from-file=key.asc=assets/gpg/private2.gpg
+    - name: helm-secrets-private-keys
+      secret:
+        secretName: helm-secrets-private-keys
   volumeMounts:
     - mountPath: /custom-tools
       name: custom-tools
     - mountPath: /usr/local/sbin/helm
       subPath: helm
       name: custom-tools
-
+    - mountPath: /helm-secrets-private-keys/
+      name: helm-secrets-private-keys
   initContainers:
     - name: download-tools
       image: alpine:latest
+      imagePullPolicy: IfNotPresent
       command: [sh, -ec]
       env:
         - name: HELM_SECRETS_VERSION
@@ -251,7 +261,8 @@ repoServer:
         - |
           mkdir -p /custom-tools/helm-plugins
           wget -qO- https://github.com/jkroepke/helm-secrets/releases/download/v${HELM_SECRETS_VERSION}/helm-secrets.tar.gz | tar -C /custom-tools/helm-plugins -xzf-;
-
+          
+          wget -qO /custom-tools/curl https://github.com/moparisthebest/static-curl/releases/latest/download/curl-amd64
           wget -qO /custom-tools/sops https://github.com/mozilla/sops/releases/download/v${SOPS_VERSION}/sops-v${SOPS_VERSION}.linux
           wget -qO /custom-tools/kubectl https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl
 
@@ -303,7 +314,7 @@ server:
 
 # Configuration of ArgoCD
 
-When using private key encryption, it is required to configure ArgoCD repo server so that it has access 
+When using private key encryption, it is required to configure ArgoCD repo server so that it has access
 to the private key to decrypt the encrypted value file(s). When using GCP KMS, encrypted value file(s)
 can be decrypted using [Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials).
 
@@ -354,7 +365,7 @@ before running sops. Define `SOPS_AGE_RECIPIENTS` is only required on initial en
 
 ### Creating the kubernetes secret holding the exported private key
 ```bash
-kubectl create secret generic helm-secrets-private-keys --from-file=key.asc
+kubectl -n argocd create secret generic helm-secrets-private-keys --from-file=key.asc=assets/gpg/private2.gpg
 ```
 
 ### Making the key accessible within ArgoCD
@@ -366,9 +377,9 @@ This is an example values file for the [ArgoCD Server Helm chart](https://argopr
 ```yaml
 repoServer:
   env:
-  - name: HELM_SECRETS_LOAD_GPG_KEYS
-    # Multiple keys can be separated by space
-    values: /helm-secrets-private-keys/key.asc
+    - name: HELM_SECRETS_LOAD_GPG_KEYS
+      # Multiple keys can be separated by space
+      value: /helm-secrets-private-keys/key.asc
   volumes:
     - name: helm-secrets-private-keys
       secret:
@@ -413,12 +424,12 @@ repoServer:
     name: argocd-repo-server
 
   rbac:
-  - apiGroups:
-    - ""
-    resources:
-    - secrets
-    verbs:
-    - get
+    - apiGroups:
+        - ""
+      resources:
+        - secrets
+      verbs:
+        - get
 ```
 
 RBAC permissions can be verified by executing the command below:
@@ -450,9 +461,9 @@ spec:
 
 sops and vals are supporting multiple cloud providers.
 
-### AWS 
+### AWS
 
-The argocd-repo-server need access to cloud services. If ArgoCD is deployed on an EKS, 
+The argocd-repo-server need access to cloud services. If ArgoCD is deployed on an EKS,
 [AWS IRSA](https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html) can be used here.
 
 This is an example values file for the [ArgoCD Server Helm chart](https://argoproj.github.io/argo-helm):
@@ -469,7 +480,7 @@ repoServer:
 If IRSA is not available, move forward with static credentials.
 
 1. Create a secret contain the `AWS_DEFAULT_REGION`, `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
-   Example: 
+   Example:
    ```bash
    kubectl create secret generic argocd-aws-credentials \
      --from-literal=AWS_DEFAULT_REGION=eu-central-1 \
@@ -488,8 +499,8 @@ If IRSA is not available, move forward with static credentials.
 
 ### GCP KMS
 
-To work with GCP KMS encrypted value files, no private keys need to be provided to ArgoCD, 
-but the Kubernetes ServiceAccount which runs the argocd-repo-server needs to have the `cloudkms.cryptoKeyVersions.useToDecrypt` permission. 
+To work with GCP KMS encrypted value files, no private keys need to be provided to ArgoCD,
+but the Kubernetes ServiceAccount which runs the argocd-repo-server needs to have the `cloudkms.cryptoKeyVersions.useToDecrypt` permission.
 There are various ways to achieve this,
 but the recommended way is to use [GKE Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
 Please read Google's documentation to link your Kubernetes ServiceAccount and a Google Service Account.

--- a/docs/ArgoCD Integration.md
+++ b/docs/ArgoCD Integration.md
@@ -76,7 +76,7 @@ References:
 * https://github.com/argoproj/argo-cd/pull/11966
 
 
-On ArgoCD 2.6.x, `sops` isn't supported in Multi-Source application, because the source reference, e.g.: `$ref` needs to be at the beginn of a string.
+On ArgoCD 2.6.x, helm-secrets isn't supported in Multi-Source application, because the source reference, e.g.: `$ref` needs to be at the beginn of a string.
 This is in conflict with helm-secrets, since the string needs to beginn with `secrets://`. On top, ArgoCD do not resolve references in URLs.
 
 `HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH` must be set to `true`, since ArgoCD pass value files with absolute file path.

--- a/docs/ArgoCD Integration.md
+++ b/docs/ArgoCD Integration.md
@@ -75,7 +75,6 @@ References:
 * https://github.com/argoproj/argo-cd/issues/11866
 * https://github.com/argoproj/argo-cd/pull/11966
 
-
 On ArgoCD 2.6.x, helm-secrets isn't supported in Multi-Source application, because the source reference, e.g.: `$ref` needs to be at the beginn of a string.
 This is in conflict with helm-secrets, since the string needs to beginn with `secrets://`. On top, ArgoCD do not resolve references in URLs.
 
@@ -85,8 +84,12 @@ Ensure that the env `HELM_SECRETS_WRAPPER_ENABLED=true` (default `false`) and
 `HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH=true` is set on the argocd-repo-server.
 Please ensure you are following the lastest installation instructions (updated on 2023-03-03).
 
-If you are using `sops` backend, additionally define the environment variable `HELM_SECRETS_LOAD_GPG_KEYS` with the path of gpg key as values.
+### sops backend
+
+If you are using `sops` backend, you have to [mounte](#method-1--mount-the-private-key-from-a-kubernetes-secret-as-volume-on-the-argocd-repo-server)
+the gpg keys on the `argocd-repo-server` and additionally define the environment variable `HELM_SECRETS_LOAD_GPG_KEYS` with the path of gpg key as values.
 Read more about mounting gpg keys [here](#method-1--mount-the-private-key-from-a-kubernetes-secret-as-volume-on-the-argocd-repo-server)
+
 
 **Note**: The limitation lives on ArgoCD side. helm-secrets is not able to mitigate the limitations at all.
 

--- a/docs/ArgoCD Integration.md
+++ b/docs/ArgoCD Integration.md
@@ -79,8 +79,11 @@ References:
 On ArgoCD 2.6.x, `sops` isn't supported in Multi-Source application, because the source reference, e.g.: `$ref` needs to be at the beginn of a string.
 This is in conflict with helm-secrets, since the string needs to beginn with `secrets://`. On top, ArgoCD do not resolve references in URLs.
 
-Ensure that the env `HELM_SECRETS_WRAPPER_ENABLED=true` (default `false`) is set on the argocd-repo-server. 
-Please ensure you are following the lastest installation instructions (updated on 2023-02-18).
+`HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH` must be set to `true`, since ArgoCD pass value files with absolute file path.
+
+Ensure that the env `HELM_SECRETS_WRAPPER_ENABLED=true` (default `false`) and 
+`HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH=true` is set on the argocd-repo-server. 
+Please ensure you are following the lastest installation instructions (updated on 2023-03-03).
 
 If you are using `sops` backend, additionally define the environment variable `HELM_SECRETS_LOAD_GPG_KEYS` with the path of gpg key as values.
 Read more about mounting gpg keys [here](#method-1--mount-the-private-key-from-a-kubernetes-secret-as-volume-on-the-argocd-repo-server)
@@ -150,7 +153,7 @@ ENV HELM_SECRETS_BACKEND="vals" \
     HELM_SECRETS_VALUES_ALLOW_SYMLINKS=false \
     HELM_SECRETS_VALUES_ALLOW_ABSOLUTE_PATH=false \
     HELM_SECRETS_VALUES_ALLOW_PATH_TRAVERSAL=false \
-    HELM_SECRETS_WRAPPER_ENABLED=true
+    HELM_SECRETS_WRAPPER_ENABLED=false
 
 # Optionally, set default gpg key for sops files
 # ENV HELM_SECRETS_LOAD_GPG_KEYS=/path/to/gpg.key
@@ -165,9 +168,6 @@ RUN apt-get update && \
 RUN curl -fsSL https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl \
     -o /usr/local/bin/kubectl && chmod +x /usr/local/bin/kubectl
 
-# -- UNCOMMENT LINE for support multi source apps
-RUN printf '#!/usr/bin/env sh\nif [ "\${HELM_SECRETS_WRAPPER_ENABLED}" = "true" ]; then exec %1\$s secrets "$@"; fi\nexec %1\$s "$@"' "helm" "${HELM_SECRETS_HELM_PATH}" >"/usr/local/sbin/helm" && chmod +x "/custom-tools/helm"
-
 # sops backend installation (optional)
 RUN curl -fsSL https://github.com/mozilla/sops/releases/download/v${SOPS_VERSION}/sops-v${SOPS_VERSION}.linux \
     -o /usr/local/bin/sops && chmod +x /usr/local/bin/sops
@@ -176,6 +176,8 @@ RUN curl -fsSL https://github.com/mozilla/sops/releases/download/v${SOPS_VERSION
 RUN curl -fsSL https://github.com/helmfile/vals/releases/download/v${VALS_VERSION}/vals_${VALS_VERSION}_linux_amd64.tar.gz \
     | tar xzf - -C /usr/local/bin/ vals \
     && chmod +x /usr/local/bin/vals
+
+RUN ln -sf "$(helm env HELM_PLUGINS)/helm-secrets/scripts/wrapper/helm.sh" /usr/local/sbin/helm
 
 USER argocd
 
@@ -209,6 +211,8 @@ repoServer:
       value: /custom-tools/kubectl
     - name: HELM_SECRETS_CURL_PATH
       value: /custom-tools/curl
+    - name: HELM_SECRETS_HELM_PATH
+      value: /usr/local/bin/helm
     - name: HELM_SECRETS_BACKEND
       value: "vals" # or sops
     # https://github.com/jkroepke/helm-secrets/wiki/Security-in-shared-environments
@@ -220,8 +224,6 @@ repoServer:
       value: "false"
     - name: HELM_SECRETS_WRAPPER_ENABLED
       value: "false"
-    - name: HELM_SECRETS_HELM_PATH
-      value: /usr/local/bin/helm
   volumes:
     - name: custom-tools
       emptyDir: {}
@@ -255,7 +257,7 @@ repoServer:
 
           wget -qO- https://github.com/helmfile/vals/releases/download/v${VALS_VERSION}/vals_${VALS_VERSION}_linux_amd64.tar.gz | tar -xzf- -C /custom-tools/ vals;
 
-          printf '#!/usr/bin/env sh\nif [ "${HELM_SECRETS_WRAPPER_ENABLED}" = "true" ]; then exec %1$s secrets "$@"; fi\nexec %1$s "$@"' "helm" "${HELM_SECRETS_HELM_PATH}" >"/usr/local/sbin/helm" && chmod +x "/custom-tools/helm"
+          cp /custom-tools/helm-plugins/helm-secrets/scripts/wrapper/helm.sh /custom-tools/helm
           
           chmod +x /custom-tools/*
       volumeMounts:

--- a/scripts/commands/downloader.sh
+++ b/scripts/commands/downloader.sh
@@ -12,19 +12,6 @@ KEY_LOCATION_PREFIX="${HELM_SECRETS_KEY_LOCATION_PREFIX:-""}"
 # shellcheck source=scripts/commands/decrypt.sh
 . "${SCRIPT_DIR}/commands/decrypt.sh"
 
-_trap_kill_gpg_agent() {
-    if [ -n "${_GNUPGHOME+x}" ]; then
-        if [ -f "${_GNUPGHOME}/.helm-secrets" ]; then
-            # On CentOS 7, there is no kill option
-            case $(gpgconf --help 2>&1) in
-            *--kill*)
-                gpgconf --kill gpg-agent
-                ;;
-            esac
-        fi
-    fi
-}
-
 downloader() {
     # https://helm.sh/docs/topics/plugins/#downloader-plugins
     # It's always the 4th parameter
@@ -186,9 +173,7 @@ _gpg_init() {
     _GNUPGHOME=$(_mktemp -d)
     touch "${_GNUPGHOME}/.helm-secrets"
 
-    GNUPGHOME="${_GNUPGHOME}"
-    export GNUPGHOME
-
+    export GNUPGHOME="${_GNUPGHOME}"
     gpg --batch --no-permission-warning --quiet --import "${1}"
 }
 

--- a/scripts/lib/common.sh
+++ b/scripts/lib/common.sh
@@ -46,8 +46,15 @@ _trap() {
         _trap_hook
     fi
 
-    if command -v _trap_kill_gpg_agent >/dev/null; then
-        _trap_kill_gpg_agent
+    if [ -n "${_GNUPGHOME+x}" ]; then
+        if [ -f "${_GNUPGHOME}/.helm-secrets" ]; then
+            # On CentOS 7, there is no kill option
+            case $(gpgconf --help 2>&1) in
+            *--kill*)
+                gpgconf --kill gpg-agent
+                ;;
+            esac
+        fi
     fi
 
     rm -rf "${TMPDIR}"
@@ -66,23 +73,21 @@ _mktemp() {
 }
 
 _gpg_load_keys() {
-    export GNUPGHOME="${GNUPGHOME:-"${HOME}/.secrets"}"
-    if [ ! -d "${GNUPGHOME}" ]; then
-        mkdir -p "${GNUPGHOME}"
-        set -x
+    _GNUPGHOME=$(_mktemp -d)
+    touch "${_GNUPGHOME}/.helm-secrets"
 
-        for key in ${LOAD_GPG_KEYS}; do
-            if [ -d "${key}" ]; then
-                set +f
-                for file in "${key%%/}/"*; do
-                    gpg --batch --no-permission-warning --quiet --import "${file}"
-                done
-                set -f
-            else
-                gpg --batch --no-permission-warning --quiet --import "${key}"
-            fi
-        done
-    fi
+    export GNUPGHOME="${_GNUPGHOME}"
+    for key in ${LOAD_GPG_KEYS}; do
+        if [ -d "${key}" ]; then
+            set +f
+            for file in "${key%%/}/"*; do
+                gpg --batch --no-permission-warning --quiet --import "${file}"
+            done
+            set -f
+        else
+            gpg --batch --no-permission-warning --quiet --import "${key}"
+        fi
+    done
 }
 
 on_wsl() { false; }

